pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -350,6 +350,12 @@ public static class QueryOptionKey {
 
         public static final String DROP_RESULTS = "dropResults";
 
+        // Handle JOIN Overflow
+
+        public static final String MAX_ROWS_IN_JOIN = "maxRowsInJoin";
+        public static final String MAX_BYTES_IN_JOIN = "maxBytesInJoin";
+        public static final String JOIN_OVERFLOW_MODE = "joinOverflowMode";
```

Comment:
Don't add them as query options, add them as query hint under `joinOptions`. See `PinotHintOptions` for more examples

Reply:
Will do, was also thinking of using hints, just thinking if we really need per join operator level control.

Reply:
Fair point. We should come up with some best practice on when to use hint and when to use option